### ISSUE TITLE
Operator v4.1.3 continuously updates tenant state from "Initialized" to "Provisioning Console Deployment" and back

### ISSUE THREAD

**Issue by dnskr** [minio/operator#715]
Tenant state continuously changes from "Initialized" to "Provisioning Console Deployment" and back if Operator v4.1.3 and minio-operator:4.1.7 chart is used

## Expected Behavior
"Provisioning Console Deployment" -> "Initialized"

## Current Behavior
"Provisioning Console Deployment" -> "Initialized" -> "Provisioning Console Deployment" -> "Initialized" -> ...

## Steps to Reproduce (for bugs)
**Install operator v4.1.3 by minio-operator 4.1.7 chart**
1. install helm chart minio-operator:4.1.7 - issue occurs

**Upgrading operator v4.1.2 to v4.1.3 by upgrading chart minio-operator 4.1.6 to 4.1.7**
1. install helm chart minio-operator:4.1.6 - no issue
2. upgrade helm chart to minio-operator:4.1.7 - issue occurs
3. set version back to minio-operator:4.1.6 - no issue

Tenant manifest with state
```yaml
  apiVersion: minio.min.io/v2
  kind: Tenant
  metadata:
    name: minio
    namespace: minio-tenant
  spec:
    console:
      image: minio/console:v0.7.5
      replicas: 1
      resources: {}
    image: minio/minio:RELEASE.2021-06-17T00-10-46Z
    imagePullPolicy: IfNotPresent
    mountPath: /export
    podManagementPolicy: Parallel
    pools:
    - affinity: {}
      nodeSelector: {}
      resources: {}
      servers: 1
      tolerations: []
      volumeClaimTemplate:
        metadata:
          name: data
        spec:
          accessModes:
          - ReadWriteOnce
          resources:
            requests:
              storage: 15Gi
          storageClassName: standard
      volumesPerServer: 4
  status:
    availableReplicas: 1
    certificates:
      autoCertEnabled: false
    currentState: Provisioning Console Deployment
    drivesOnline: 4
    healthStatus: green
    pools:
    - ssName: minio-ss-0
      state: PoolInitialized
    revision: 0
    syncVersion: ""
    writeQuorum: 3
```

**Comment by dnskr** [comment]
_Might be related to the problem_

If docker image updated to **RELEASE.2021-07-08T19-43-25Z** than in console log I see
```
Status: 4 Online, 0 Offline.
API: http://172.17.0.6:9000 http://127.0.0.1:9000
Console: http://172.17.0.6:38921 http://127.0.0.1:38921
Documentation: https://docs.min.io
WARNING: Console endpoint is listening on a dynamic port (38921), please use --console-address ":PORT" to choose a static port.
```

There is no this WARNING for **RELEASE.2021-06-17T00-10-46**Z docker image

**Comment by dnskr** [comment]
Related [issue#716](https://github.com/minio/operator/issues/716)

**Comment by agilob** [comment]
An ingress pointing to a minio service causes redirect from `myminiodomain.com` redirect to `myminiodomain.com:<some high port>` rendering

**Comment by harshavardhana** [comment]
@agilob use the default image that comes with the operator - the operator does not yet support the latest MinIO release. 

You can turn-off embedded console with in MinIO with `MINIO_BROWSER=off`

**Comment by agilob** [comment]
I'm just trying minio out using `docker run` [command from website](https://docs.min.io/docs/minio-docker-quickstart-guide.html), then on a raspberry pi I have k8s minio-svc and ingress that manage cert-manager for me and forward myminiodomain.com to my local pc.

The command on website uses `minio/minio` which means "latest" tag and this one doesn't work as it causes redirects on ports that arent forwarded. what version should I use then? The versioned image I posted above has different WebUI than the one with redirect breaking access from outside my LAN.

**Comment by harshavardhana** [comment]
> I'm just trying minio out using `docker run` [command from website](https://docs.min.io/docs/minio-docker-quickstart-guide.html), then on a raspberry pi I have k8s minio-svc and ingress that manage cert-manager for me and forward `https://myminiodomain.com` to my local pc.

if you do that then make sure to set MINIO_BROWSER_REDIRECT_URL - https://github.com/minio/minio#test-using-minio-console 

are you not even reading these and doing blind upgrades?

**Comment by agilob** [comment]
>doing blind upgrades?

I'm installing it first time on a disposable machine

**Comment by dnskr** [comment]
> @agilob use the default image that comes with the operator - the operator does not yet support the latest MinIO release.
> 
Hi @harshavardhana,
Does operator still not support latest MinIO release? What's needed to be done to release a new version of operator and helm chart for it? I'd glad to contribute if possible.

**Comment by harshavardhana** [comment]
> Does operator still not support latest MinIO release? What's needed to be done to release a new version of operator and helm chart for it? I'd glad to contribute if possible.

We have already done the work waiting on few more last changes @dnskr - thank you for the helping hand - feel free to contribute to the master branch if there are things that are missing.

**Comment by dnskr** [comment]
Fixed with Operator v4.2.0 release
